src/connection_string.ts

```diff
@@ -542,7 +545,7 @@ export function parseOptions(
     loggerClientOptions
   );
 
-  return mongoOptions;
+  return { ...mongoOptions, ...internalLoggerConfig };
```

Comment:
Instead of making changes here, a code path that will always be reached, can we instead make 
`internalLoggerConfig` spread into the `loggerClientOptions` only? That way the functionality is still only in the flagged code path